src/main.js

```diff
@@ -16,60 +17,108 @@ const checkValidationOfValues = envSpecString => {
+    else {
       checkValidation = false;
     }
   });
 
   //in case of one or more invalid variables or types, return error
   if (checkValidation === true) {
-    return envValArray;
+    return envSpecEntries;
   } else {
     return null;
   }
 };
 
-//function that separates the variables from their types and returns them as a two-dimensional array
+//function that separates the variables from their types and returns them as an array
 const parseVarFromType = envSpecAsArray => {
-  return (envSpecAsArray = envSpecAsArray.map(element => {
+  return (envSpecAsArrayParsed = envSpecAsArray.map(element => {
+    //in case of typed variable or given restricted choices
     if (element.includes(":")) {
       element = element.split(":");
-      return [element[0].trim(), element[1].trim()];
-    } else {
-      //if untyped environmental variable set default as text
-      return [element.trim(), "text"];
+      let entry = {
+        name: element[0].trim(),
+        type: element[1].trim(),
+        choices: null
+      };
+      return entry;
+    }
+    //in case of untyped variable , give default type
+    else {
+      let entry = {
+        name: element.trim(),
+        type: "text",
+        choices: null
+      };
+      return entry;
     }
   }));
 };
 
+//function that creates the render label for the output outputHTML
+const renderLabelForEntry = name => {
+  return (labelToPrint = `<label for="env_spec_${name.toLowerCase()}">${name}</label>\n`);
+};
+
 //function that returns the HTML code as a string
-//envValues is a two-dimensional array containing the variable name and its type
-const outputHTML = envValues => {
-  //create HTML format
-  if (envValues) {
-    envValues = envValues.map(
-      element =>
-        `<label for="env_spec_${element[0].toLowerCase()}">${
-          element[0]
-        }</label>\n` +
-        `<input id="env_spec_${element[0].toLowerCase()}" name="${element[0].toLowerCase()}" type="${
-          element[1]
-        }" />\n`
-    );
+//envSpecEntriesArray is an array containing entries with their variable name and its type or restricted choices
+const outputHTML = envSpecEntriesArray => {
+  //create HTML format if there is not a syntax error
+  //in case of syntax error envSpecEntriesArray is null
+  if (envSpecEntriesArray) {
+    envSpecEntriesToPrint = envSpecEntriesArray.map(element => {
+      toPrint = renderLabelForEntry(element.name);
+      //if element has valid type
+      if (element.type) {
+        toPrint += `<input id="env_spec_${element.name.toLowerCase()}" name="${element.name.toLowerCase()}" type="${
+          element.type
+        }" />\n`;
+        return toPrint;
+      } else if (element.choices) {
+        //if element is value with restricted choices
+        toPrint += `<select id="env_spec_${element.name.toLowerCase()}" name="${element.name.toLowerCase()}">\n`;
+        for (let i = 0; i < element.choices.length; i++) {
```

Comment:
Change the `for` loop please to one without counters 😄:

```javascript
for (const choice of element.choices) {
    // Do things with choice, instead of `element.choices[i]`. Much cleaner!
}
```